### PR TITLE
[PM-33561] debt: Wire CipherManager and cipher ViewModel error handlers

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-33561

## 📔 Objective

Wire `userFriendlyMessage` through CipherManager `fold(onFailure)` blocks and update all cipher-related ViewModel error handlers to prefer `errorMessage` over generic defaults.

This is part of the PM-33394 series to surface `CookieRedirectException` user-friendly messages through result types to ViewModels. The foundation (`userFriendlyMessage` extension and `errorMessage` on result types) was merged in #6642.

**Changes:**
- Wire `userFriendlyMessage` in CipherManager for delete, restore, share, archive, unarchive, and attachment operations
- Update VaultItemViewModel, VaultAddEditViewModel, AttachmentsViewModel, VaultMoveToOrganizationViewModel, SearchViewModel, VaultItemListingViewModel, and VaultViewModel error handlers
- Add CipherManager tests for CookieRedirectException error message propagation
- Add ViewModel tests for non-null errorMessage display

### CODE REVIEW

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt [hunk 1]
```diff
@@ -169,7 +180,8 @@ class CipherManagerImpl(
         cipherId: String,
         cipherView: CipherView,
     ): ArchiveCipherResult {
-        val userId = activeUserId ?: return ArchiveCipherResult.Error(NoActiveUserException())
+        val userId = activeUserId
+            ?: return ArchiveCipherResult.Error(error = NoActiveUserException())
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt:170]: Why did this change?

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt [hunk 2]
```diff
@@ -196,15 +208,21 @@ class CipherManagerImpl(
...
             )
     }
 
     override suspend fun unarchiveCipher(
         cipherId: String,
         cipherView: CipherView,
     ): UnarchiveCipherResult {
-        val userId = activeUserId ?: return UnarchiveCipherResult.Error(NoActiveUserException())
+        val userId = activeUserId
+            ?: return UnarchiveCipherResult.Error(error = NoActiveUserException())
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt:206]: Ditto

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt (hunk 1 above)
Comment (SaintPatrck) [reply app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt:170]: Leftover from adding `errorMessage` because I originally added it as the first param so the param name had to be added. Just forgot to revert these two spots.

They're minor changes and we usually declare param names anyways. Unless you feel strongly about leaving it as it was, I think this actually keeps things more consistent with the rest of the code.

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt (hunk 2 above)
Comment (SaintPatrck) [reply app/src/main/kotlin/com/x8bit/bitwarden/data/vault/manager/CipherManagerImpl.kt:206]: https://github.com/bitwarden/android/pull/6651#discussion_r2930984173

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/CreateCipherResult.kt [hunk 3]
```diff
@@ -14,5 +16,8 @@ sealed class CreateCipherResult {
      * Generic error while creating cipher. The optional [errorMessage] may be displayed directly in
      * the UI when present.
      */
-    data class Error(val errorMessage: String?, val error: Throwable?) : CreateCipherResult()
+    data class Error(
+        val error: Throwable? = null,
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/CreateCipherResult.kt]: Should this have a default? I think most (maybe all) cases supply a throwable

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/UpdateCipherResult.kt [hunk 4]
```diff
@@ -14,5 +16,8 @@ sealed class UpdateCipherResult {
      * Generic error while updating cipher. The optional [errorMessage] may be displayed directly in
      * the UI when present.
      */
-    data class Error(val errorMessage: String?, val error: Throwable?) : UpdateCipherResult()
+    data class Error(
+        val error: Throwable? = null,
```
Comment (david-livefront) [inline app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/UpdateCipherResult.kt]: Same question here

File: app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/CreateCipherResult.kt (hunk 3 above)
Comment (SaintPatrck) [reply app/src/main/kotlin/com/x8bit/bitwarden/data/vault/repository/model/CreateCipherResult.kt]: Nope. Reverted. Good catch.
